test/deprecated/RegexpModule.chpl

```diff
@@ -1,9 +0,0 @@
-use Regexp;                                      // deprecated module
-try! {
-    var re = Regexp.compile("*");
-} catch err : BadRegexpError {                   // deprecated type
-    writeln(err.message());
-}
-var re : regexp = Regexp.compile("a*");          // deprecated type
-var match : reMatch = re.match("aaaaaaaaaaa");   // deprecated type
-writeln(match.matched);
```

Comment:
Since your PR does more than just updates for the environment variable, it would be good to mention what else it does in the PR message and final commit message.  For instance, it should mention that you are also removing the Regexp module deprecation test

Reply:
> Since your PR does more than just updates for the environment variable, it would be good to mention what else it does in the PR message and final commit message. For instance, it should mention that you are also removing the Regexp module deprecation test

I am not removing the `Regexp` module deprecation test. Git reports files/directories as "deleted" if they are both renamed and modified. There is another corresponding file which was added, so the deletion + addition = renaming + slightly modied

The directory name was changed to `regex` instead of `regexp`, since the `regex` name is favored instead of `regexp`, but the `Regexp` deprecation test remains.